jerry-libm/jerry-libm-internal.h

```diff
@@ -30,14 +30,25 @@
 /* Sometimes it's necessary to define __LITTLE_ENDIAN explicitly
    but these catch some common cases. */
 
-#if (defined (i386) || defined (__i386) || defined (__i386__) || \
+
+#ifndef __LITTLE_ENDIAN
+/* Check if compiler has byte order macro. Some older versions do not.
+ * If byte order is supported and set to little define __LITTLE_ENDIAN.
+ */
+#ifdef __BYTE_ORDER__
+#if (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
+#define __LITTLE_ENDIAN
+/* Common cases are checked when byte order macro is not supported. */
+#elif (defined (i386) || defined (__i386) || defined (__i386__) || \
      defined (i486) || defined (__i486) || defined (__i486__) || \
      defined (intel) || defined (x86) || defined (i86pc) || \
      defined (__alpha) || defined (__osf__) || \
      defined (__x86_64__) || defined (__arm__) || defined (__aarch64__) || \
-     defined (__xtensa__))
+     defined (__xtensa__) || defined (__MIPSEL))
```

Comment:
Can't we just add the `(defined (__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__))` after the MIPSEL macro? In addition: do we need to check the `__LITTLE_ENDIAN` at the top?

Reply:
`(defined (__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__))` after MIPSEL sure, no problem. I have just put the broadest check first. 
`#ifndef __LITTLE_ENDIAN` check was needed with previous version, now it is not necessary but if macro is defined anywhere before (like command line) build will fail so I left it.

Reply:
I have changed order of checks and removed update of mipsel toolchain but left` #ifndef __LITTLE_ENDIAN`.

Reply:
Hi, is there anything that needs to be changed in code for it to be merged?

Reply:
First of all the CI must be green, please check the logs: https://travis-ci.org/pando-project/jerryscript/jobs/514555761

Reply:
Warnings are for something this pull request doesn't change. I'll see to it. Thank you.

Reply:
Files with error are changed 2 years ago and errors are:
```
jerry-libm/sqrt.c:237: error(uninitvar): Uninitialized variable: z
jerry-libm/trig.c:736: error(uninitvar): Uninitialized variable: qx
```
but this is same for all other checks but no errors are reported.
I will correct it but I am little confused by this.

Reply:
I will take a look at on this later. @galpeter any idea why did these issues appear?

Reply:
My first guess would be that the cppcheck incorrectly detects the macro expansion. I'll check it on my pc.

Reply:
yeah... cppcheck can't properly process the new byte_order check and will incorrectly assume that the __LO/__HI calls are functions....

Reply:
@lidija-b by reworking the parentheses a bit the cppcheck now correctly processes the files, could you also check it?

Here's the diff (note the closing parenthesis after the MIPSEL check) :
```diff
 #if (defined (i386) || defined (__i386) || defined (__i386__) || \
      defined (i486) || defined (__i486) || defined (__i486__) || \
      defined (intel) || defined (x86) || defined (i86pc) || \
      defined (__alpha) || defined (__osf__) || \
      defined (__x86_64__) || defined (__arm__) || defined (__aarch64__) || \
-     defined (__xtensa__))
+     defined (__xtensa__) || defined (__MIPSEL)) || \
+     (defined (__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__))
 #define __LITTLE_ENDIAN
 #endif
```

Reply:
Looks good to me, pushed with suggested change

Reply:
looks good, but sadly the vera checker complains that the byteorder check line does have an indentation. We need to remove the whitespaces before that line.

Reply:
I rushed to push, sorry. Corrected, pushed and passed checks.